bedrock-codec/src/main/java/org/cloudburstmc/protocol/bedrock/codec/v419/serializer/StartGameSerializer_v419.java

```diff
@@ -178,6 +178,15 @@ protected void readLevelSettings(ByteBuf buffer, BedrockCodecHelper helper, Star
         packet.setForceExperimentalGameplay(helper.readOptional(buffer, OptionalBoolean.empty(), buf -> OptionalBoolean.of(buf.readBoolean())));
     }
 
+    protected PlayerPermission readPlayerPermission(ByteBuf buffer) {
+        final int ordinal = VarInts.readInt(buffer);
+        if (ordinal == -1) {
```

Comment:
I think it would make more sense to check for actual array bounds instead of checking for only `-1`.